File: src/main/kotlin/com/melqui/serverkotlinpool/controller/pool/Pool.java
package com.melqui.serverkotlinpool.controller.pool;

import com.melqui.serverkotlinpool.dto.CreatePoolDTO;
import com.melqui.serverkotlinpool.services.PoolServices;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import javax.validation.Valid;

@RestController
@RequestMapping("pools")
class Pool {
    @Autowired
    private PoolServices poolServices;
    @PostMapping
    public void createPool(@RequestBody @Valid CreatePoolDTO createPoolDTO) {

        this.poolServices.create(createPoolDTO);
    }
    @GetMapping
    public Iterable<com.melqui.serverkotlinpool.model.Pool> listPools() {
        return this.poolServices.list();
    }
}

File: src/main/kotlin/com/melqui/serverkotlinpool/controller/pool/Pool.kt
package com.melqui.serverkotlinpool.controller.pool

import com.melqui.serverkotlinpool.dto.CreatePoolDTO
import com.melqui.serverkotlinpool.services.PoolServices
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequestMapping("pools")
class Pool(private var poolService: PoolServices) {
    @PostMapping
    fun createPool(@RequestBody @Valid createPoolDTO: CreatePoolDTO) {
        poolService.create(createPoolDTO)
    }
}